### ISSUE TITLE
Autoconf and pure perl pkg-config interactions under windows

### ISSUE THREAD

**Issue by shawnlaffan** [PerlAlien/Alien-Build#227]
This is a follow-on from discussion in #195 (see https://github.com/PerlAlien/Alien-Build/pull/195#issuecomment-657329357 ).

There are colon issues with drive names and path separators when passing between Windows and MSYS, where the pure perl pkg-config is in use.  Wrapping the pure perl pkg-config is one means of avoiding these issues, which was the purpose of #195 .  Unfortunately the shell wrapper in #195 did not handle quoted arguments.  

I have reworked the pure-perl pkg-config wrapper approach using perl for the wrapper script to better handle quoted arguments.  I've released it to CPAN as a plugin, with development on GitHub.

https://metacpan.org/release/Alien-Build-Plugin-PkgConfig-PPWrapper
https://github.com/shawnlaffan/Alien-Build-Plugin-PkgConfig-PPWrapper

It works for Alien::gdal and Alien::proj based on CI results (see https://github.com/shawnlaffan/geo-gdal-ffi-canary , noting that recent CI failures are unrelated to the plugin code).

If you decide it's worth including in Alien::Build then I can work up another PR, or feel free to adapt as needed.

**Comment by plicease** [comment]
Thanks for doing this work!

I think including the plugin in AB core would be an option.  I've done this before with other plugins, though none that we're written by me yet.  I usually let them stew for a while to collect cpantesters results, but I've noticed this plugin doesn't have any tests aside from the basic "this module can compile".  So I'd like to see a few tests firsts `Test::Alien::Build` can be useful here.

Another option would be to have this built into the AutoConf plugin.  If it is "on by default" that benefits the most users, unless there are packages that it breaks, which makes me nervous of course.  If it is opt-in then it isn't functionally much better than if it is a separate plugin.

**Comment by shawnlaffan** [comment]
No worries.  

You are right about the tests, and this is listed as shawnlaffan/Alien-Build-Plugin-PkgConfig-PPWrapper#1 
I was not sure how to set something up to actually test the functionality but will have a look at ```Test::Alien::Build```.  

I now use it in CPAN releases of Alien::gdal and Alien::proj and it is working thus far, so there is at least indirect testing.  The cpan-testers failures are unrelated.  
https://metacpan.org/release/SLAFFAN/Alien-gdal-1.23
https://metacpan.org/release/SLAFFAN/Alien-proj-1.15

**Comment by plicease** [comment]
> I now use it in CPAN releases of Alien::gdal and Alien::proj and it is working thus far, so there is at least indirect testing. The cpan-testers failures are unrelated.
> https://metacpan.org/release/SLAFFAN/Alien-gdal-1.23
> https://metacpan.org/release/SLAFFAN/Alien-proj-1.15

Yup this is good :+1:  I still want some unit tests for anything in AB core to make it easy to refactor stuff and catch things early on.

**Comment by zmughal** [comment]
I don't know if this will help, but has anybody else looked at:

- `pkg-config-lite`: <https://sourceforge.net/projects/pkgconfiglite/> ( installable using <https://community.chocolatey.org/packages/pkgconfiglite>)
- `u-config`: <https://github.com/skeeto/u-config> (described at <https://nullprogram.com/blog/2023/01/18/> )

I have used `pkg-config-lite` successfully with Strawberry Perl 5.38.
